src/main/java/org/elasticsearch/index/mapper/object/RootObjectMapper.java

```diff
@@ -284,15 +302,25 @@ protected void doXContent(XContentBuilder builder, ToXContent.Params params) thr
             }
         }
 
-        if (dynamicTemplates != null && dynamicTemplates.length > 0) {
-            builder.startArray("dynamic_templates");
+        if (dynamicTemplates != null && !dynamicTemplates.isEmpty()) {
+            boolean nonEmptyTemplatePresent = false;
             for (DynamicTemplate dynamicTemplate : dynamicTemplates) {
-                builder.startObject();
-                builder.field(dynamicTemplate.name());
-                builder.map(dynamicTemplate.conf());
-                builder.endObject();
+                nonEmptyTemplatePresent = !dynamicTemplate.isEmpty();
+            }
+            
+            if (nonEmptyTemplatePresent) {
+                builder.startArray("dynamic_templates");
+                for (DynamicTemplate dynamicTemplate : dynamicTemplates) {
+                    if(dynamicTemplate.isEmpty()) {
+                        continue;
+                    }
+                    builder.startObject();
+                    builder.field(dynamicTemplate.name());
+                    builder.map(dynamicTemplate.conf());
+                    builder.endObject();
+                }
+                builder.endArray();
```

Comment:
Why do we need to check for empty templates here? It should never happen since we only merge non-empty templates?